pkg/kubelet/container/runtime.go

```diff
@@ -52,6 +52,7 @@ type Version interface {
 // information about the different image types.
 type ImageSpec struct {
 	Image string
+	Pod *v1.Pod
```

Comment:
What is the driving reason to have Pod info in Image spec? Usually image is not pod specific.

Reply:
this is a fair ask. for now ( maybe forever ), we won't support mixed workload type so the info used to identify the image service in the API  chain is *pod -- this is mainly to simplify the way to get the desired image service -- using POD info is probably the most straightforward way with the current APIs.